unit_test/environment_test.py

```diff
@@ -37,12 +38,15 @@ def test_inheritance():
 def test_shell_eval():
     environment_recipe = parse_environment('VAR="$(echo "a test" string)"')
 
+    env_copy = os.environ.copy()
+    env_copy.pop('VAR', None)
+
     environment_dict = environment_recipe.as_dictionary(
-        prev_environment={}
+        prev_environment=env_copy
     )
     environment_cmds = environment_recipe.as_shell_commands()
 
-    assert environment_dict == {'VAR': 'a test string'}
+    assert environment_dict['VAR'] == 'a test string'
```

Comment:
What's the thinking/reasoning behind this change?

Reply:
ah, i see it fixes Python 3.8 on azure... what's your thinking why?

Reply:
I checked the build log. a bug in python 3.8?

Reply:
It's either a bug in python 3.8 on windows or a bug in the tests.
For sure, behavior has changed which would suggest a bug in python 3.8 however passing an empty environment to subprocess feels wrong to me so I updated the test to pass the full environment instead.